RecoLocalTracker/Phase2TrackerRecHits/plugins/Phase2StripCPEESProducer.cc

```diff
@@ -0,0 +1,34 @@
+
+#include "RecoLocalTracker/Phase2TrackerRecHits/plugins/Phase2StripCPEESProducer.h"
+#include "RecoLocalTracker/Phase2TrackerRecHits/interface/Phase2StripCPEDummy.h"
+
+
+Phase2StripCPEESProducer::Phase2StripCPEESProducer(const edm::ParameterSet & p) {
+  std::string name = p.getParameter<std::string>("ComponentType");
+
+  enumMap_[std::string("Phase2StripCPEDummy")] = DUMMY;
+  if (enumMap_.find(name) == enumMap_.end())
+    throw cms::Exception("Unknown StripCPE type") << name;
+
+  cpeNum_ = enumMap_[name];
+  pset_ = p;
+  setWhatProduced(this, name);
+}
+
+
+boost::shared_ptr<ClusterParameterEstimator<Phase2TrackerCluster1D> > Phase2StripCPEESProducer::produce(const TkStripCPERecord & iRecord) {
```

Comment:
https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/FWCore/Skeletons/scripts/mkTemplates/ESProducer/ESProducer.cc
is the template (updated in 2014) incorrect?